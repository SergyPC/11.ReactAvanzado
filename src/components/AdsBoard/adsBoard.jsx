import React, { Component } from 'react';
// import { BrowserRouter as Router, Route, Link, Switch, withRouter } from "react-router-dom";
import imgNoDisponible from '../../img/noDisponible.jpg'
import { CardColumns }  from 'react-bootstrap';
import Navbar from '../Navbar/navbar';
import AdFilter from '../AdFilter/adFilter';
import CardAd from '../CardAd/cardAd';

export default class AdsBoard extends Component {
    constructor(props) {
        super(props);
        // this.state = {
        //     search: sessionStorage.getItem('search') ? sessionStorage.getItem('search') : '',
        //     data: [],
        //     tags: [],
        //     params: {
        //         name: sessionStorage.getItem('name') ? sessionStorage.getItem('name') : '',
        //         minPrice: sessionStorage.getItem('minPrice') ? sessionStorage.getItem('minPrice') : '',
        //         maxPrice: sessionStorage.getItem('maxPrice') ? sessionStorage.getItem('maxPrice') : '',
        //         venta: sessionStorage.getItem('venta') ? sessionStorage.getItem('venta') : '',
        //         tag: sessionStorage.getItem('tag') ? sessionStorage.getItem('tag') : '',
        //     }
        // };
    }

    componentDidMount() {
        this.props.getUserFromStorage();
        this.props.fetchAds();
        this.props.fetchTags();
    }

    render() {
        return (
            <div>
                <Navbar />
                <AdFilter tags={this.props.tags} props={this.props} />
                <AdList ads={this.props.ads} props={this.props} /> 
            </div>
        )
    }
}

function AdList ({ads}) {
    return(
        <div>
            <CardColumns>
                {ads.map(card => {
                    //const photo = ((card.photo.indexOf("http://") > -1) || (card.photo.indexOf("https://") > -1)) ? card.photo : 'http://www.sogarca.com/wp-content/uploads/2015/06/No-disponible.jpg';
                    card.photo = ((card.photo.indexOf("http://") > -1) || (card.photo.indexOf("https://") > -1)) ? card.photo : imgNoDisponible;
                        return (<CardAd key={card._id} card={card} />);
                        // return (
                        //     <Card key={card._id}>
                        //         <Card.Img variant="top" src={card.photo} />
                        //         <Card.Body>
                        //             <Link to={`/adsBoard/${card._id}`}>
                        //                 <Card.Title>{card.name}</Card.Title>
                        //             </Link>
                        //             <Card.Text>
                        //                 Type: {card.type}
                        //                 <br />
                        //                 Price: {card.price}â‚¬
                        //                 <br /><br />
                        //                 Description:
                        //                 <br />
                        //                 {card.description}
                        //             </Card.Text>
                        //         </Card.Body>
                        //         <Card.Footer>
                        //             <small className="text-muted">Last updated: {card.updatedAt}</small>
                        //             <Link to={`/editAd/id=${card._id}`}>
                        //                 <Button variant="success" size="lg" block>
                        //                     Edit Ad
                        //                 </Button>
                        //             </Link>
                        //         </Card.Footer>
                        //     </Card>
                        // );
                    })
                }
            </CardColumns>
        </div>
    )
};
