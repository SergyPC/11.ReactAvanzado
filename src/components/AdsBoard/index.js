import AdsBoard from './adsBoard';
import { connect } from 'react-redux';
import { fetchAds, fetchTags, getUserFromStorage } from '../../store/actions';
import { getAds, getTags, isLogged } from '../../store/selectors';

function mapStateToProps(state, ownProps) {
  return {
    ads: getAds(state),
    tags: getTags(state),
    user: isLogged(state),
  }
}

const mapDispatchToProps = {
  fetchAds,
  fetchTags,
  getUserFromStorage
}

export default connect(mapStateToProps, mapDispatchToProps)(AdsBoard);
