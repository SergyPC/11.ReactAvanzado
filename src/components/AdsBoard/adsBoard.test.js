import React from 'react';
import { shallow } from 'enzyme';
import AdsBoard from './AdsBoard';

describe('Snapshot testing.', ()=> {
    const props = {
        ads: [
            {
                _id: '5e4af71d3976de16b4d34276',
                name: 'Alfa Romeo',
                price: 1500,
                description: 'Lo vendo, baratito baratito pisha',
                type: 'sell',
                photo: 'https://cdn.pixabay.com/photo/2020/02/24/06/15/castle-4875322__340.jpg',
            },
            {
                _id: '5e4c3b443976de16b4d34287',
                name: 'Fallout New Vegas 2',
                price: 10,
                description: 'PS3 version',
                type: 'buy',
                photo: 'https://upload.wikimedia.org/wikipedia/en/3/34/Fallout_New_Vegas.jpg',
            },
        ],
        tags: ['lifestyle', 'mobile', 'motor', 'work'],
        user: true,
        fetchAds: jest.fn(),
        getUserFromStorage: jest.fn(),
        fetchTags: jest.fn(),
    };

    let wrapper;

    beforeEach(()=> {
        wrapper = shallow(<AdsBoard {...props} />);
    });

    test("Renderizado de componentes", ()=> {
        expect(wrapper.exists('AdList')).toBe(true)
    });

    test("Debe renderizar una lista de anuncios", ()=> {
        //expect(wrapper.find('AdList').props().ads).toEqual(props.ads);
        expect(wrapper.find('AdList').props().ads).toHaveLength(2); //props.ads.length
    });

    test('Snapshot de AdsBoard', ()=> {
        expect(wrapper).toMatchSnapshot();
    });

});