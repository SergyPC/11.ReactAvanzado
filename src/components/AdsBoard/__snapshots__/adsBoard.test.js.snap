// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing. Snapshot de AdsBoard 1`] = `ShallowWrapper {}`;
